### PR TITLE
Crud

### PR DESCRIPTION
**Pull request by AshirMohab**
Very jank fix to reloading state

### CODE REVIEW

**Comment by AshirMohab** [comment]
Added lazy loading as well as state management for the currency api. The site is somewhat responsive as well

File: src/app/services/user.service.ts [hunk 1]
```diff
@@ -16,14 +16,6 @@ export class UserService {
...
 
-  async addUserTrip(trip: TripsModel) {
-    this.angularFireStore.collection('Trips').add({
-      ...trip,
-      userID: await this.angularFireAuth.currentUser.then((user) => user?.uid),
-      tripID: this.angularFireStore.createId(),
-    });
-  }
-
   async deleteUser(userID: string) {
```
Comment (theadley) [inline src/app/services/user.service.ts:19]: Is this used? By what effect? async await is not compatible with state management.

File: src/app/store/currency/currency.selectors.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  createSelector,
+  props,
+} from '@ngrx/store';
+import { CurrencyData, CurrencyType } from 'src/app/models/currency';
+import * as fromCurrency from './currency.reducer';
+
+export const selectCurrencyState =
+  createFeatureSelector<fromCurrency.CurrencyState>(
+    fromCurrency.currencyFeatureKey
+  );
```
Comment (theadley) [inline src/app/store/currency/currency.selectors.ts:13]: It looks like you need to still add the selectors for all of the currencies and the selected currency

File: src/app/store/trip/trip.effects.ts [hunk 3]
```diff
@@ -1,43 +1,140 @@
...
-  tripTrips$ = createEffect(() => {
+  getUserTrips$ = createEffect(() => {
     return this.actions$.pipe(
-      ofType(TripActions.getUserTrips),
-      concatMap(() =>
-        this.tripService.getUserTrips().pipe(
+      ofType(TripActions.getUserTrips, setCurrenctUser),
+      withLatestFrom(this.userStore.pipe(select(selectCurrentUser))),
+      concatMap(([action, currentUser]) => {
+        // const currentUser = { uid: 'yDYYtFyxd3QqkA9FTevxOOC7Suy2' };
```
Comment (theadley) [inline src/app/store/trip/trip.effects.ts:28]: Commented out code to be removed please

File: src/app/store/trip/trip.effects.ts [hunk 4]
```diff
@@ -1,43 +1,140 @@
...
+        )
+      )
+    );
+  });
+
+  removeUserTrip$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(TripActions.removeUserTrip),
+      concatMap(({ tripToRemove }) =>
+        this.tripService.editUserTrip(tripToRemove).pipe(
```
Comment (theadley) [inline src/app/store/trip/trip.effects.ts:114]: Why is removeUserTrip calling editUserTrip? Doesn't it need the tripId first?

File: src/app/store/trip/trip.selectors.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { createFeatureSelector, createSelector } from '@ngrx/store';
-import { selectTripItineraries } from '../itinerary/itinerary.selectors';
+// import { selectTripItineraries } from '../itinerary/itinerary.selectors';
```
Comment (theadley) [inline src/app/store/trip/trip.selectors.ts:2]: Commented out code to be removed please

**Review by theadley — CHANGES_REQUESTED** [review]
Looks like a great PR. It's a bit long for a PR, so keep that in mind.
The majority of my comments are for removing commented out code, so be mindful of not committing commented out code. Git is your friend if you need to dig into the past.

File: src/app/components/itinerary/my-itineraries/my-itineraries.component.ts [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+  @Input() itinerary!: ItineraryItem[] | null;
+  @Input() currency!: string | undefined;
 
-  constructor() { }
-
-  ngOnInit(): void {
-  }
+  constructor() {}
 
+  ngOnInit(): void {}
```
[the remark this answers is not in the document]
Comment (AshirMohab) [reply src/app/components/itinerary/my-itineraries/my-itineraries.component.ts:15]: This just takes in the list of itinerary items of a trip and then will display them

File: src/app/lazy/lazy.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-lazy',
+  templateUrl: './lazy.component.html',
+  styleUrls: ['./lazy.component.scss']
+})
+export class LazyComponent implements OnInit {
```
[the remark this answers is not in the document]
Comment (AshirMohab) [reply src/app/lazy/lazy.component.ts:8]: I know but it was just helping me know what was up

File: src/app/services/auth.service.ts [hunk 8]
```diff
@@ -77,45 +90,17 @@ export class AuthService {
...
-  //           .doc(`Users/${user.uid}`)
-  //           .set({ ...user }, { merge: true });
-  //       }
-  //     })
-  //     .catch((err: Error) => {
-  //       console.error(err.message);
-  //     });
-  // }
-
   get isLoggedIn(): boolean {
```
[the remark this answers is not in the document]
Comment (AshirMohab) [reply src/app/services/auth.service.ts:93]: Just checked it, doesn't seem to constantly run

**Review by theadley — APPROVED** [review]
Good soup

**Pull request merged**